tools/pipelines-tasks/common/helpers.ts

```diff
@@ -116,4 +122,38 @@ export const writeXml = (xmlObject: any, filePath: string) =>
 {
     const xmlBuilder = new xml.Builder();
     fs.writeFileSync(filePath, xmlBuilder.buildObject(xmlObject));
+}
+
+/**
+ * installs input Nuget package into given output Location.
+ * @param packagePath the path to the package(.nupkg or packages.config) file to be installed
+ * @param outputPath output location to install the nuget package into
+ * @param targetNetFramework target .Net framework to pick target and dependency DLLs complying with the package to be installed
+ * @param packageId optional parameter, package id to be installed. Required if package install mechanism is through providing package id(instead of packages.config)
+ * @param version optional parameter, package version to be installed. Required if package install mechanism is through providing package id(instead of packages.config)
+ */
+export async function installNuget(packagePath: string, outputPath: string, targetNetFramework: string, packageId?: string, version?: string) {
```

Comment:
As I see, common is used as a helper that can be used in common to all tasks. It would help to have this in common although the current purpose of it is during build time. Say, we have a dependency in future where we have to install some nuget package on runtime dynamically, for example, say it is for building an msix package for some specific project type or pulling some dependencies for app attach etc..., it might help in having this here.

Reply:
While it _might_ happen that you need to pull some nuget package dynamically at runtime, I don't think that will really happen. I don't know of any other projects doing that, and I don't see why this would be different.

We also should prefer getting all dependencies at build time, so that we can track which versions we are using. That would prevent compatibility issues if a dependency were to change on us, and helps us track if there are any vulnerabilities on our dependencies.

Reply:
Got it, pulling this out from common package making sure it is not shipped or available to other tasks to avoid dynamic nuget install